profitbricks/src/main/java/org/jclouds/profitbricks/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,480 @@
+   private final ProvisioningJob.Factory jobFactory;
+   private final ProvisioningManager provisioningManager;
+
+   private static final Integer LAN_ID = 1;
+
+   @Inject
+   ProfitBricksComputeServiceAdapter(ProfitBricksApi api,
+           @Named(POLL_PREDICATE_DATACENTER) Predicate<String> waitDcUntilAvailable,
+           @Named(PROPERTY_USER_THREADS) ListeningExecutorService executorService,
+           ProvisioningJob.Factory jobFactory,
+           ProvisioningManager provisioningManager) {
+      this.api = api;
+      this.waitDcUntilAvailable = waitDcUntilAvailable;
+      this.executorService = executorService;
+      this.jobFactory = jobFactory;
+      this.provisioningManager = provisioningManager;
+   }
+
+   @Override
+   public NodeAndInitialCredentials<Server> createNodeWithGroupEncodedIntoName(String group, String name, Template template) {
+      final String dataCenterId = template.getLocation().getId();
+      Hardware hardware = template.getHardware();
+
+      TemplateOptions options = template.getOptions();
+      final String loginUser = isNullOrEmpty(options.getLoginUser()) ? "root" : options.getLoginUser();
+      final String password = options.hasLoginPassword() ? options.getLoginPassword() : Passwords.generate();
+
+      final org.jclouds.compute.domain.Image image = template.getImage();
+
+      // provision all storages based on hardware
+      List<? extends Volume> volumes = hardware.getVolumes();
+      List<String> storageIds = Lists.newArrayListWithExpectedSize(volumes.size());
+
+      int i = 1;
+      for (final Volume volume : volumes)
+         try {
+            logger.trace("<< provisioning storage '%s'", volume);
+            final Storage.Request.CreatePayload request = Storage.Request.creatingBuilder()
+                    .dataCenterId(dataCenterId)
+                    // put image to first storage
+                    .mountImageId(i == 1 ? image.getId() : "")
+                    .imagePassword(password)
+                    .name(format("%s-disk-%d", name, i++))
+                    .size(volume.getSize())
+                    .build();
+
+            String storageId = (String) provisioningManager.provision(jobFactory.create(dataCenterId, new Supplier<Object>() {
+
+               @Override
+               public Object get() {
+                  return api.storageApi().createStorage(request);
+               }
+            }));
+
+            storageIds.add(storageId);
+            logger.trace(">> provisioning complete for storage. returned id='%s'", storageId);
+         } catch (Exception ex) {
+            if (i - 1 == 1) // if first storage (one with image) provisioning fails; stop method
+               throw Throwables.propagate(ex);
+            logger.warn(ex, ">> failed to provision storage. skipping..");
+         }
+
+      Double cores = 0d;
+      for (Processor processor : hardware.getProcessors())
+         cores += processor.getCores();
```

Comment:
[minor] Use the [ComputeServiceUtils](https://github.com/jclouds/jclouds/blob/master/compute/src/main/java/org/jclouds/compute/util/ComputeServiceUtils.java#L116-L121) instead.